src/com/github/Kraken3/AFKPGC/BotDetector.java

```diff
@@ -115,6 +116,21 @@ public synchronized void doDetector() {
 		currentTPS = TpsReader.getTPS();
 		AFKPGC.debug("Bot Detector Running, TPS is: ", currentTPS);
 		Map<UUID, LastActivity> lastActivities = LastActivity.lastActivities;
+		if (currentTPS < startingTPS) {
+			for (Map.Entry<UUID, LastActivity> entry : lastActivities.entrySet()) {
+				UUID playerUUID = entry.getKey();
+				Player p = Bukkit.getPlayer(playerUUID);
+				if (lastActivities.containsKey(playerUUID) && p != null) {
+					LastActivity la = entry.getValue();
+					la.loggedLocations.add(p.getLocation());
+					if (la.loggedLocations.size() >= maxLocations) {

```

Comment:
You're testing the same thing twice; just use ```if (la.loggedLocations.size() > maxLocations) {``` and get rid of the line 126 check.